File: benchmark-tests/benchmark.js
function cons (x, y) {this.x = x; this.y = y}

require ('sys').puts ('cons1');
for (var i = 0; i < 8000000; ++i) new cons (3, 5);

require ('sys').puts ('cons2');
var conses = [];
for (var i = 0; i < 8000000; ++i) {
  conses.push (3);
  conses.push (5);
}

File: benchmark-tests/benchmark.rb
#!/usr/bin/ruby

def cons1 x, y
  [x, y]
end

Conses = []
def cons2 x, y
  Conses << x << y
end

1000000.times {|i| cons1 3, 5}
puts "cons1"

1000000.times {|i| cons2 3, 5}
puts "cons2"
